crates/turborepo-lib/src/package_graph/builder.rs

```diff
@@ -94,22 +101,58 @@ impl<'a> PackageGraphBuilder<'a> {
         self
     }
 
+    #[allow(dead_code)]
+    pub fn with_package_discovery<P2: PackageDiscovery>(
+        self,
+        discovery: P2,
+    ) -> PackageGraphBuilder<'a, P2> {
+        PackageGraphBuilder {
+            repo_root: self.repo_root,
+            root_package_json: self.root_package_json,
+            is_single_package: self.is_single_package,
+            package_manager: self.package_manager,
+            package_jsons: self.package_jsons,
+            lockfile: self.lockfile,
+            package_discovery: discovery,
+        }
+    }
+}
+
+impl<'a> PackageGraphBuilder<'a, ()> {
+    /// Build the `PackageGraph` with the default package discovery strategy.
     #[tracing::instrument(skip(self))]
-    pub fn build(self) -> Result<PackageGraph, Error> {
+    pub async fn build_default(self) -> Result<PackageGraph, Error> {
+        let builder = PackageGraphBuilder {
+            package_discovery: LocalPackageDiscovery::new(self.repo_root.to_owned())?,
```

Comment:
I believe we're already performing double package manager discovery with this call and then the `PackageManager::get_package_manager` call in `BuildState::new`

Reply:
Ok, after a bit of thinking, best strategy imo is to have the package manager be a required input, that way we can create the LocalPackageDiscovery without wasting time looking for the manager.